src/scripts/main.js

```diff
@@ -1,3 +1,285 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('.button');
+const fieldCells = document.querySelectorAll('.field_cell');
+const gameScore = document.querySelector('.game-score');
+const messageLose = document.querySelector('.message_lose');
+const messageWin = document.querySelector('.message_win');
+const messageStart = document.querySelector('.message_start');
+let board = [];
+let score = 0;
+let playGame = true;
+let mergedCells = [];
+
+function getRandomNumber() {
+  return Math.random() < 0.9 ? 2 : 4;
+};
+
+function generateRandomNumbers() {
+  let gameOver = true;
+
+  for (let row = 0; row < 4; row++) {
+    for (let col = 0; col < 4; col++) {
+      if (board[row][col] === 0) {
+        gameOver = false;
+      }
+    }
+  }
+
+  if (gameOver) {
+    messageLose.classList.remove('hidden');
+  } else {
+    for (let i = 0; i < 1; i++) {
+      const row = Math.floor(Math.random() * 4);
+      const col = Math.floor(Math.random() * 4);
+
+      if (board[row][col] === 0) {
+        board[row][col] = getRandomNumber();
+      } else {
+        i--;
+      }
+    }
+  }
+};
+
+function createField() {
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  mergedCells = [
+    [false, false, false, false],
+    [false, false, false, false],
+    [false, false, false, false],
+    [false, false, false, false],
+  ];
+
+  generateRandomNumbers(board);
+  generateRandomNumbers(board);
+};
+
+function mergeAndMove(row, col, rowChange, colChange) {
+  if (board[row][col] !== 0) {
+    let kRow = row + rowChange;
+    let kCol = col + colChange;
+
+    while (kRow >= 0 && kRow < 4 && kCol >= 0
+      && kCol < 4 && board[kRow][kCol] === 0) {
+      kRow += rowChange;
+      kCol += colChange;
+    }
+
+    if (kRow >= 0 && kRow < 4 && kCol >= 0
+      && kCol < 4 && board[kRow][kCol] === board[row][col]
+      && !mergedCells[kRow][kCol]) {
```

Comment:
Such long expression divide by shorter and move to variables with meaningful variable naming